src/libcore/alloc/layout.rs

```diff
@@ -258,19 +258,45 @@ impl Layout {
 
     /// Creates a layout describing the record for `self` followed by
     /// `next`, including any necessary padding to ensure that `next`
-    /// will be properly aligned. Note that the resulting layout will
-    /// satisfy the alignment properties of both `self` and `next`.
+    /// will be properly aligned, but *no trailing padding*. In order to
+    /// match C representation layout, you should call `pad_to_align`
```

Comment:
This also applies to Rust representation layout, right? (Just that there, additionally the field order is unspecified.)

This could be read as saying that you don't need to worry about this for `repr(Rust)`.

Reply:
I worry about mentioning `repr(Rust)` here, because `repr(Rust)` is entirely (deliberately) unspecified. Rustc is free to add frivolous padding everywhere if it wanted, in addition to "just" reordering fields.

Because of this, it's not really meaningful to manipulate a `Layout` at all in respect to `repr(Rust)`.

`GlobalAlloc`/`AllocRef` do not have an intrinsic requirement that size is a multiple of alignment, so saying `pad_to_align` must always be called would be incorrect.

Reply:
Maybe be explicit about that then, in a parenthetical:
> (There is no way to match the default Rust representation layout `repr(Rust)`, as it is unspecified.)

Also explicitly mentioning `repr(C)` could be helpful to make the connection.
> In order to match C representation layout `repr(C)`, ...

Reply:
I've added text to that effect.